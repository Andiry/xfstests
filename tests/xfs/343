#! /bin/bash
# FS QA Test No. 343
#
# Basic rmap manipulation tests for realtime files.
#
#-----------------------------------------------------------------------
# Copyright (c) 2016, Oracle and/or its affiliates.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------

seq=`basename "$0"`
seqres="$RESULT_DIR/$seq"
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1    # failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -rf "$tmp".*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here
_supported_os Linux
_supported_fs xfs
_require_realtime
_require_xfs_scratch_rmapbt
_require_xfs_io_command "falloc"
_require_xfs_io_command "fpunch"
_require_xfs_io_command "fzero"
_require_xfs_io_command "fcollapse"
_require_xfs_io_command "finsert"

rm -f "$seqres.full"

echo "Format and mount"
_scratch_mkfs > "$seqres.full" 2>&1
_scratch_mount
blksz=65536
blocks=16
len=$((blocks * blksz))

echo "Create some files"
$XFS_IO_PROG -f -R -c "falloc 0 $len" -c "pwrite -S 0x68 -b 1048576 0 $len" $SCRATCH_MNT/f1 >> $seqres.full

echo "Manipulate file"
$XFS_IO_PROG -c "fpunch $blksz $blksz" \
	-c "fzero $((3 * blksz)) $blksz" \
	-c "pwrite -S 0x69 $((5 * blksz)) $blksz" \
	-c "fpunch $((7 * blksz)) $blksz" \
	-c "fsync" \
	-c "pwrite -S 0x70 $((7 * blksz)) $blksz" \
	-c "fcollapse $((9 * blksz)) $blksz" \
	-c "finsert $((10 * blksz)) $blksz" $SCRATCH_MNT/f1 >> $seqres.full

echo "Check file"
md5sum $SCRATCH_MNT/f1 | _filter_scratch
od -tx1 -Ad -c $SCRATCH_MNT/f1 >> $seqres.full

# success, all done
status=0
exit
