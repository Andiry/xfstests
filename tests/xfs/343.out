QA output created by 343
Format and mount
Create some files
Manipulate file
Check file
e45c5707fcf6817e914ffb6ce37a0ac7  SCRATCH_MNT/f1
